Added ray version of the html2parquet transform

sungeunan-ibm opened the pull request:
## Why are these changes needed?
Added ray version of the html2parquet transform

## Related issue number (if any).
Issue #635

Commit:
moved to language folder

Signed-off-by: Sungeun An <sungeun.an@ibm.com>

touma-I committed:
Merge branch 'dev' into html2parquet

touma-I committed:
Seperated requirements from pyproject.toml

Signed-off-by: Maroun Touma <touma@us.ibm.com>

touma-I committed:
added requirements.txt

Signed-off-by: Maroun Touma <touma@us.ibm.com>

touma-I committed:
Apply folder change to ci/cd

Signed-off-by: Maroun Touma <touma@us.ibm.com>

Commit:
merge from dev

Commit:
Add ray version of html2parquet

shahrokhDaijavad commented:
Great. Thank you very much @sungeunan-ibm for finishing this. It looks like the conflict in yml is the result of 1) The move of this transform from universal to the language folder and 2) The fact that a new hap transform was added yesterday. But, the main reviewer for the correct implementation of ray is @daw3rd.

sungeunan-ibm commented:
Thank you @shahrokhDaijavad! @daw3rd Could you advise on how I should resolve these conflicts? Should I simply regenerate the file by running the following commands?
```
cd .github/workflows
rm test-language-html2parquet.yml 
make
```

daw3rd commented on .github/workflows/test-language-html2parquet.yml:
```diff
@@ -1,7 +1,11 @@
 #
 # DO NOT EDIT THIS FILE: it is generated from test-transform.template,  Edit there and run make to change these files
 #
+<<<<<<<< HEAD:.github/workflows/test-language-html2parquet.yml
```
How did you check this in?  I thought git would not allow it.  Anyway.  I recommend the following.
1. merge the latest dev branch into your branch
2. run `make` in this directory and you should get a proper version of this file.

daw3rd commented on .github/workflows/test-universal-hap.yml:
```diff
@@ -1,7 +1,11 @@
 #
 # DO NOT EDIT THIS FILE: it is generated from test-transform.template,  Edit there and run make to change these files
 #
+<<<<<<<< HEAD:.github/workflows/test-language-html2parquet.yml
```
Same here

daw3rd commented on transforms/language/html2parquet/ray/Makefile:
```diff
@@ -0,0 +1,62 @@
+
+minio-start:	.minio-start
+
+kind-load-image:: .transforms.kind-load-image
```
Please update the following rules to match those in other transforms from the latest dev branch.  For example, https://github.com/IBM/data-prep-kit/blob/eadce898542f00493b7c0cc6cdde04d86dc7be6b/transforms/universal/ededup/Makefile#L56

sungeunan-ibm replied:
The Makefile you are referring to is located in the ray folder, while the example provided is directly under the ededup folder. I checked the Makefile for the ededup transform within the ray folder, and it matches the same structure.

daw3rd commented on transforms/language/html2parquet/ray/README.md:
```diff
@@ -0,0 +1,17 @@
+
+The transform will produce the following statsd metrics:
+
+| metric name                      | Description                                                      |
```
These are not part of the output parquet file.  I think you need to tell the reader where these can be viewed.   Also, please link from here to the python readme for annotations and configuration.

sungeunan-ibm replied:
Actually, I'm not entirely sure where these metrics can be viewed. In the Ray implementation, I borrowed the pdf2parquet code, but I’m not very familiar with Prometheus metrics. These metrics are generated from the `Html2ParquetRayTransform` class, and they can be viewed through logging. After some investigation, it appears that you can access these metrics in the Prometheus dashboard under the relevant namespace for your application. However, my experience with this is limited. I would greatly appreciate your assistance in adding more detailed descriptions!

Commit:
Regenerated workflow files and edited Makefile

Signed-off-by: Sungeun An <sungeun.an@ibm.com>

Commit:
added python readme

Signed-off-by: Sungeun An <sungeun.an@ibm.com>

Commit:
resolve conflicts

Signed-off-by: Sungeun An <sungeun.an@ibm.com>

Commit:
resolved conflicts

Signed-off-by: Sungeun An <sungeun.an@ibm.com>

touma-I commented on transforms/language/html2parquet/ray/pyproject.toml:
```diff
@@ -0,0 +1,46 @@
+[project]
+name = "dpk_html2parquet_transform_python"
```
```suggestion
name = "dpk_html2parquet_transform_ray"
```

touma-I commented on transforms/language/html2parquet/ray/Makefile:
```diff
@@ -0,0 +1,62 @@
+
+include $(REPOROOT)/transforms/.make.transforms
+
+venv::	.transforms.python-venv
```
```suggestion
venv::	.transforms.ray-venv
```

touma-I commented on transforms/language/html2parquet/ray/Makefile:
```diff
@@ -0,0 +1,62 @@
+
+venv::	.transforms.python-venv
+
+test::	.transforms.python-test
```
```suggestion
test::	.transforms.ray-test
```

touma-I commented on transforms/language/html2parquet/ray/Makefile:
```diff
@@ -0,0 +1,62 @@
+
+clean:: .transforms.clean
+
+image:: .transforms.python-image
```
```suggestion
image:: .transforms.ray-image
```

touma-I commented on transforms/language/html2parquet/ray/Makefile:
```diff
@@ -0,0 +1,62 @@
+
+publish: publish-image
+
+publish-image:: .transforms.publish-image-python
```
```suggestion
publish-image:: .transforms.publish-image-ray
```

touma-I commented on transforms/language/html2parquet/ray/Makefile:
```diff
@@ -0,0 +1,62 @@
+
+publish-dist:: .defaults.publish-dist
+
+test-image:: .transforms.python-test-image
```
```suggestion
test-image:: .transforms.ray-test-image
```

touma-I commented on transforms/language/html2parquet/ray/Makefile:
```diff
@@ -0,0 +1,62 @@
+	    --data_local_config \" { 'input_folder' : '../test-data/input', 'output_folder' : '../output' } \"  \
+	    --data_files_to_use \"['.html','.zip']\"  
+
+run-local-sample: .transforms.run-local-sample
```
```suggestion
run-local-sample: .transforms.run-local-ray-sample
```

touma-I commented on transforms/language/html2parquet/ray/Makefile:
```diff
@@ -0,0 +1,62 @@
+
+run-local-sample: .transforms.run-local-sample
+
+run-local-python-sample: .transforms.run-local-python-sample
```
```suggestion
```

touma-I committed:
Update transforms/language/html2parquet/ray/pyproject.toml

package name for ray cannot be the same as package name for python

touma-I committed:
Update transforms/language/html2parquet/ray/Makefile

Setting up the vent for ray testing, is different than python testing

touma-I committed:
Update transforms/language/html2parquet/ray/Makefile

here we need to test ray modules (not python)

touma-I committed:
Update transforms/language/html2parquet/ray/Makefile

Need to publish ray image (not python)

touma-I committed:
Update transforms/language/html2parquet/ray/Makefile

Need to build ray image (not python)

touma-I committed:
Update transforms/language/html2parquet/ray/Makefile

Need to test ray image (not python)

touma-I committed:
Update transforms/language/html2parquet/ray/Makefile

Not sure if we are using this but it should still be ray sample as the default goes to python sample

touma-I committed:
Update transforms/language/html2parquet/ray/Makefile

This won't work here

touma-I commented on .github/workflows/test-code-license_select-kfp.yml:
```diff
@@ -102,6 +102,7 @@ jobs:
             - name: Test KFP libs (shared and v2) and run a workflow
               timeout-minutes: 120
               run: |
+
```
Why is this file being changed by this PR?

touma-I commented on transforms/language/html2parquet/Makefile:
```diff
@@ -53,7 +53,7 @@ docker-save-image::
 	$(MAKE) RULE=$@ .recurse
 
-.PHONY: workflow-venv
+.PHONY: workflow-vent
```
There is no kfp support yet. Why is this in here ?

daw3rd committed:
restore test-code-license_select-kfp.yml

Signed-off-by: David Wood <dawood@us.ibm.com>

touma-I commented on transforms/language/html2parquet/ray/Makefile:
```diff
@@ -0,0 +1,61 @@
+TRANSFORM_NAME= html2parquet
+
+include $(REPOROOT)/transforms/.make.transforms
+
```
```suggestion
BASE_IMAGE=${RAY_BASE_IMAGE}
```

touma-I committed:
Update transforms/language/html2parquet/ray/Makefile

Need a base image to build the docker image for ray

touma-I committed:
Merge branch 'dev' into html2parquet
